test/tests/postprocessors/num_time_steps/moose.i

```diff
@@ -0,0 +1,98 @@
+[Mesh]
```

Comment:
There is no need for a complex solve to test this. The test should be as minimal as possible to satisfy the requirement listed in the test spec:

> The shall be able to count the number of time-steps that have been taken over the course of the simulation of a transient

A coupled heat conduction solve and a large 27k element mesh is absolutely overkill for testing this postprocessor.

A basic problem without a solve is sufficient:

```
[Mesh/gmg]
  type = GeneratedMeshGenerator
  dim = 1
[]

[Postprocessors/timestep_ctr]
  type = NumTimeSteps
[]

[Problem]
  solve = False
[]

[Executioner]
  type = Transient
  dt = 1
  num_steps = 10
[]

[Outputs]
  csv = true
[]
```